### ISSUE TITLE
Calling startActivity() from outside of an Activity context requires the FLAG_ACTIVITY_NEW_TASK flag

### ISSUE THREAD

**Issue by sohaib37104** [RodrigoSMarques/flutter_branch_sdk#266]
**Describe the bug**
Clicking on link and going to app through chrome working fine but i when  i am going through "App Links that  open URLs directly into an app" is giving this errror.
E/MethodChannel#flutter_branch_sdk/message(14070): Failed to handle method call
E/MethodChannel#flutter_branch_sdk/message(14070): android.util.AndroidRuntimeException: Calling startActivity() from outside of an Activity  context requires the FLAG_ACTIVITY_NEW_TASK flag. Is this really what you want?
E/MethodChannel#flutter_branch_sdk/message(14070): Failed to handle method call
E/MethodChannel#flutter_branch_sdk/message(14070): android.util.AndroidRuntimeException: Calling startActivity() from outside of an Activity  context requires the FLAG_ACTIVITY_NEW_TASK flag. Is this really what you want?

**To Reproduce**
1. Intalize flutter branc sdk in flutter project 
2. create short url and share on whatsapp and then click on the link then dialog will appear where you want to open this link either through chrome or your app when i click on my app then this error is coming if i am going to open link through chrom its working fine. 

**Expected behavior**
Deep link need to go to specific screen in my app when i go thorught my app not chrome.

**Screenshots**

<img width="1291" alt="Screenshot 2023-11-29 at 2 55 52 PM" src="https://github.com/RodrigoSMarques/flutter_branch_sdk/assets/127841537/11f67e80-86be-4b00-89f2-4d627ce3aa90">

![WhatsApp Image 2023-11-29 at 2 57 43 PM](https://github.com/RodrigoSMarques/flutter_branch_sdk/assets/127841537/1435307a-21bc-45f7-8dab-369d55807549)

<img width="1438" alt="Screenshot 2023-11-29 at 3 01 49 PM" src="https://github.com/RodrigoSMarques/flutter_branch_sdk/assets/127841537/bebcec02-7319-4066-9fb5-7aa5687bfc75">

**Smartphone (Please complete the following information. remove session if not platform):**
 - Flutter: version: [e.g.3.13.0]
 - Package version: [e.g.flutter_branch_sdk: ^7.0.1]
 - OS: [Android 11]
 - Device: [Realme C25]
 

 - **Additional context**
 - i have configured the branch dashboard configuration i am sharing the screenshot of branch dashboard.
 - i hope you understand my situration if you need more information about issue let me know thanks.
 - waiting for your feedback.

**Comment by RodrigoSMarques** [comment]
problem under analysis here #261

**Comment by RodrigoSMarques** [comment]
Hi @sohaib37104 

I fixed the bug, but I need help testing the changes in different scenarios.

Can you help me with testing the new version before publishing it on pub.dev?

Update `pubspec.yaml`, according to the example below:

```dart
#flutter_branch_sdk:
flutter_branch_sdk:
     git:
       url: https://github.com/RodrigoSMarques/flutter_branch_sdk.git
       ref: v7.0.2-dev
```

I'm waiting for the test results.

Thanks.

**Comment by sohaib37104** [comment]
okay sure let me check

**Comment by sohaib37104** [comment]
Hi @RodrigoSMarques  I have check now in my application its working perfectly fine Thank you very much  for listening to my request i greatly appreciate your effort.thanks

**Comment by RodrigoSMarques** [comment]
Fixed in version 7.0.2
